ingestors/jsonschema.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+from typing import NamedTuple
+
+"""
+This module provides a flat Location structure so scraper authors do not need
+to understand the normalized ndjson spec:
+
+https://docs.google.com/document/d/1qxABDlRep76llrXhgwdwA6Ji3Jub-MftRvustk12XHc
+"""
+
+
+class Location(NamedTuple):
+    id: str = None
+    name: str = None
+    street1: str = None
+    street2: str = None
+    city: str = None
+    state: str = None
+    zip: str = None
+    latitude: str = None
+    longitude: str = None
+    phone: str = None
+    website: str = None
+    email: str = None
+    contact_other: str = None
+    provider_id: str = None
+    provider_brand: str = None
+    provider_brand_name: str = None
+    booking_website: str = None
+    booking_phone: str = None
+    booking_email: str = None
+    appointments_available: bool = False
+    fetched_at: str = None
+    published_at: str = None
+    fetched_from_uri: str = None
+    source: str = None
+    data: dict[str, str] = None
+
+
+def omit_empty(d):
+    def pop_empty_values(d):
+        for key, value in list(d.items()):
+            if isinstance(value, dict):
+                omit_empty(value)
+            elif value is None:
+                d.pop(key)
+            elif value == [None]:
+                d.pop(key)
+
+    pop_empty_values(d)
+
+    for key, value in list(d.items()):
+        if value == {}:
+            d.pop(key)
+    return d
+
+
+def to_dict(loc):
+    """ Convert Location namedtuple to nested dict """
+
+    d = {
+        "id": loc.id,
+        "name": loc.name,
+        "address": {
+            "street1": loc.street1,
+            "street2": loc.street2,
+            "city": loc.city,
+            "state": loc.state,
+            "zip": loc.zip,
+        },
+        "location": {
+            "latitude": loc.latitude,
+            "longitude": loc.longitude,
+        },
+        "contact": {
+            "phone": [loc.phone],
+            "website": [loc.website],
+            "email": [loc.email],
+            "other": [loc.contact_other],
+        },
+        "booking": {
```

Comment:
should the schema for contact and booking look identical? Seems like it may save headaches. don't feel strongly.

Reply:
Yeah, this schema is awkward. I am trying to accomplish multiple things:
1. Provide a space for collecting multiple contact info if they exist.
2. Also provide a space for label something that we know is booking information as such.

Perhaps a better schema would be:
```
"contact": [{
  "contact_type": "booking",
  "phone": "(111) 222-3333",
  "website": "https://book.the.vaccine",
  "email": "vaccine@bookings.com",
  "other": "Book in person too",
}, ...]
```